flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/MaterializedTableStatementITCase.java

```diff
@@ -236,6 +241,80 @@ void testCreateMaterializedTableInFullMode() {
+        // store
+        SessionHandle sessionHandle = initializeSession();
+
+        String materializedTableDDL =
+                "CREATE MATERIALIZED TABLE users_shops"
+                        + " PARTITIONED BY (ds)\n"
+                        + " WITH(\n"
+                        + "   'format' = 'debezium-json'\n"
+                        + " )\n"
+                        + " FRESHNESS = INTERVAL '30' SECOND\n"
+                        + " AS SELECT \n"
+                        + "  user_id,\n"
+                        + "  shop_id,\n"
+                        + "  ds,\n"
+                        + "  SUM (payment_amount_cents) AS payed_buy_fee_sum,\n"
+                        + "  SUM (1) AS pv\n"
+                        + " FROM (\n"
+                        + "    SELECT user_id, shop_id, DATE_FORMAT(order_created_at, 'yyyy-MM-dd') AS ds, payment_amount_cents FROM datagenSource"
+                        + " ) AS tmp\n"
+                        + " GROUP BY (user_id, shop_id, ds)";
+
+        OperationHandle materializedTableHandle =
+                service.executeStatement(
+                        sessionHandle, materializedTableDDL, -1, new Configuration());
+        awaitOperationTermination(service, sessionHandle, materializedTableHandle);
+
+        // check unknown partition keys
+        String alterStatementWithUnknownPartitionKey =
+                "ALTER MATERIALIZED TABLE users_shops REFRESH PARTITION (ds2 = '2023-01-01')";
+        OperationHandle alterStatementWithUnknownPartitionKeyHandle =
+                service.executeStatement(
+                        sessionHandle,
+                        alterStatementWithUnknownPartitionKey,
+                        -1,
+                        new Configuration());
+
+        assertThatThrownBy(
+                        () ->
+                                awaitOperationTermination(
+                                        service,
+                                        sessionHandle,
+                                        alterStatementWithUnknownPartitionKeyHandle))
+                .isInstanceOf(SqlExecutionException.class)
+                .rootCause()
+                .isInstanceOf(TableException.class)
+                .hasMessage("The partition spec contains unknown partition keys: [ds2].");
+
+        // check valid statement
+        long currentTime = System.currentTimeMillis();
+        String alterStatement =
+                "ALTER MATERIALIZED TABLE users_shops REFRESH PARTITION (ds = '2023-01-01')";
+        OperationHandle alterHandle =
+                service.executeStatement(sessionHandle, alterStatement, -1, new Configuration());
+        awaitOperationTermination(service, sessionHandle, alterHandle);
+        List<RowData> result = fetchAllResults(sessionHandle, alterHandle);
+        assertThat(result.size()).isEqualTo(1);
+        String jobId = result.get(0).getString(0).toString();
```

Comment:
Can you check whether this job is a Flink batch job? Can you also check the result counts?